compiler-core/src/type_.rs

```diff
@@ -1142,6 +1195,7 @@ pub fn generalise(t: Arc<Type>) -> Arc<Type> {
                 package: package.clone(),
                 name: name.clone(),
                 args,
+                constructor_index: *constructor_index,
```

Comment:
I think this may be a bug? Shouldn't generalising a type discard what variant it is?

Reply:
Oops, I must have missed that one